### ISSUE TITLE
getImage() only for selected area

### ISSUE THREAD

**Issue by developer239** [sylvain121/node-X11#1]
Hey,
awesome work! I actually started to learn C++ so that I could write simple game bot using template matching with openCv and other stuff. I did not know how to link x11 library in nodejs though so I accidentally stumbled onto your repository when I was searching the internet.

Your repository lacks documentation that is kind of sad. I had to  #look into your tests folder so that I could see how do you convert raw image data into file. Again you did it 1000 times faster than most repositories on github using turbo-jpeg that I did not really know.

I have a little problem though. I created simple screen streamer in nodejs and when you look here: https://github.com/developer239/nodejs-screen-streamer/blob/master/services/stream.js

When I take the screenshot:
```
const screenshot = this.x11.getImage()

    const options = {
      format: jpg.FORMAT_BGRA,
      width: screenshot.width,
      height: screenshot.height,
    }

    const preAllocated = new Buffer(jpg.bufferSize(options))
    const encodedScreen = jpg.compressSync(screenshot.data, preAllocated, options)
```
I am taking and compressing basically whole screen. (on ubuntu every screen that is connected for some reason) That is not efficient.

I want to use template matching with opencv in the future and I will need only about 800x700 pixels.
So I decided to crop the image:

```
    const cropParameters = {
      x: 0,
      y: 0,
      width: 800,
      height: 770
    }
    const croppedEncodedScreen = kroppa.crop(encodedScreen, cropParameters, 40)
    
    const dataUri = new Datauri()
    dataUri.format('.jpeg', croppedEncodedScreen)
    emmitImage({ dataUri: dataUri.content, time: (Date.now()) })
```
Again that is really ineficient.

Could you please help me out? How would you crop the screen lets say x: 10, y: 10, width: 800, height: 770.

**Comment by sylvain121** [comment]
Hi,
I use XshmGetimage of MIT-SHM library to grab the screen.
If you want a cropping function with this module, we must implement it with C++, because XshmGetImage has no arguments for it.

**Comment by developer239** [comment]
I switched to `robot.js` until I find out how to solve this problem. You are the guy who showed me how to use the buffer from their `robot.screen.grebScreen()` method though. Thank you so much for that. :)

I created simple image stream:
https://www.youtube.com/watch?v=0VgP0C15zMg

I wanted to do this because now I am working on simple real time image recognition (computer vision) it is not as performant as expected but it somehow works. When I have image 220x160 I can stream 60 fps to browser while searching for images. (I resized recorded area to 30% so that I can get about 600x480 pixels) Recording video would be better I guess but I don't really know how.
